solr/core/src/test/org/apache/solr/util/TestExportToolWithNestedDocs.java

```diff
@@ -0,0 +1,293 @@
+   * demonstrates creating Anonymous Children docs, and then confirming the
+   * export format.
+   */
+  public void testIndexingAnonKids() throws Exception {
+    final String collection = "test_anon";
+    CollectionAdminRequest.createCollection(collection, ANON_KIDS_CONFIG, 1, 1)
+        .setPerReplicaState(SolrCloudTestCase.USE_PER_REPLICA_STATE)
+        .process(cluster.getSolrClient());
+    cluster.getSolrClient().setDefaultCollection(collection);
+    
+    final SolrClient client = cluster.getSolrClient();
+
+    final SolrInputDocument p1 = new SolrInputDocument();
+    p1.setField("id", "P11!prod");
+    p1.setField("type_s", "PRODUCT");
+    p1.setField("name_s", "Swingline Stapler");
+    p1.setField("description_t", "The Cadillac of office staplers ...");
+    {
+      final SolrInputDocument s1 = new SolrInputDocument();
+      s1.setField("id", "P11!S21");
+      s1.setField("type_s", "SKU");
+      s1.setField("color_s", "RED");
+      s1.setField("price_i", 42);
+      { 
+        final SolrInputDocument m1 = new SolrInputDocument();
+        m1.setField("id", "P11!D41");
+        m1.setField("type_s", "MANUAL");
+        m1.setField("name_s", "Red Swingline Brochure");
+        m1.setField("pages_i", 1);
+        m1.setField("content_t", "...");
+
+        s1.addChildDocument(m1);
+      }
+
+      final SolrInputDocument s2 = new SolrInputDocument();
+      s2.setField("id", "P11!S31");
+      s2.setField("type_s", "SKU");
+      s2.setField("color_s", "BLACK");
+      s2.setField("price_i", 3);
+
+      final SolrInputDocument m1 = new SolrInputDocument();
+      m1.setField("id", "P11!D51");
+      m1.setField("type_s", "MANUAL");
+      m1.setField("name_s", "Quick Reference Guide");
+      m1.setField("pages_i", 1);
+      m1.setField("content_t", "How to use your stapler ...");
+
+      final SolrInputDocument m2 = new SolrInputDocument();
+      m2.setField("id", "P11!D61");
+      m2.setField("type_s", "MANUAL");
+      m2.setField("name_s", "Warranty Details");
+      m2.setField("pages_i", 42);
+      m2.setField("content_t", "... lifetime guarantee ...");
+
+      p1.addChildDocuments(Arrays.asList(s1, s2, m1, m2));
+    }
+    
+    client.add(p1);
+ 
+    client.commit();
+
+    String url = cluster.getRandomJetty(random()).getBaseUrl() + "/" + collection;
+    
+    String tmpFileLoc = new File(cluster.getBaseDir().toFile().getAbsolutePath() +
+        File.separator).getPath();
+    
+    ExportTool.Info info = new ExportTool.MultiThreadedRunner(url);
+    String absolutePath = tmpFileLoc + collection + random().nextInt(100000) + ".jsonl";
+    info.setOutFormat(absolutePath, "jsonl");
+    info.setLimit("-1");
+    info.query = "description_t:Cadillac";
+    info.fields = "*,[child parentFilter='type_s:PRODUCT']";
+    info.exportDocs();
+    
+    assertEquals(1, info.docsWritten.get());
+    
+    String jsonOutput = Files.readString(new File(info.out).toPath());
```

Comment:
Use Path instead of File please.  Here, just use Path.of(info.out).  Ideally info.out would already be a Path but that's maybe too much refactoring to do here now.